packages/web/src/store/application/ui/buy-audio/sagas.ts

```diff
@@ -525,12 +526,20 @@ function* purchaseStep({
   retryDelayMs,
   maxRetryCount
 }: PurchaseStepParams) {
-  // Cache current SOL balance
+  // Cache current SOL balance and tip of transaction history
   const initialBalance = yield* call(
     [connection, connection.getBalance],
     rootAccount.publicKey,
     'finalized'
   )
+  const initialTransactions = yield* call(
+    [connection, connection.getSignaturesForAddress],
+    rootAccount.publicKey,
+    {
+      limit: 1
+    }
+  )
+  const initialTransaction = initialTransactions[0].signature
```

Comment:
sorry didnt catch this one. but potentially empty array here too

Reply:
oops! i should have seen that, my b